File: HomeLibraryCore/src/main/java/com/smolderingdrake/homelibrarycore/domain/Book.java
package com.smolderingdrake.homelibrarycore.domain;

import com.smolderingdrake.homelibrarycore.model.Genre;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.ISBN;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "books")
public class Book implements Serializable {

    @Id
    @ISBN
    private String isbn;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @ManyToMany
    @JoinTable(name = "books_to_authors",
            joinColumns = @JoinColumn(name = "book_isbn"),
            inverseJoinColumns = @JoinColumn(name = "author_id"))
    private List<Author> authors;

    @Column(name = "genre")
    @Enumerated(EnumType.STRING)
    private Genre genre;

    @Column(name = "no_of_copies")
    private Integer count;
}

File: HomeLibraryWeb/src/app/book.ts
import {Author} from './author';

export class Book {
  isbn: string;
  title: string;
  authors: Author[];
  genre: string;
  count: number;
  description: string;

  convert(singleBookArray) {
    this.isbn = singleBookArray[0];
    this.title = singleBookArray[1];
    this.authors = singleBookArray[2];
    this.description = singleBookArray[3];
    this.genre = singleBookArray[4];
    this.count = singleBookArray[5];
  }
}
